engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java

```diff
@@ -646,6 +642,25 @@ yield switch (identifier) {
         var fn = translateExpression(app.getFunc(), cons(app.getArg(), moreArgs), insideTypeSignature, false);
         yield fn;
       }
+      case Tree.NamedApp app -> {
+        var fn = translateExpression(app.getFunc(), cons(app, moreArgs), insideTypeSignature, false);
+        yield fn;
+      }
+      case Tree.Array arr -> {
+        List<IR.Expression> items = nil();
+        if (arr.getFirst() != null) {
+          var exp = translateExpression(arr.getFirst(), insideTypeSignature);
```

Comment:
Shouldn't this be
```suggestion
          var exp = translateExpression(arr.getFirst().getBody(), insideTypeSignature);
```
? Or is that the reason for this getFirst vs getRest?

Reply:
I think the current state is OK. `arr.getFirst()` is just a `Tree`. Elements of `arr.getRest()` are `​OperatorDelimitedTree` and one needs to call `getBody` to get the `Tree` from them.